### PR TITLE
Replace service clippaths with a single global one

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA

Run `go run examples/generatesvg.go examples/mediawiki-scalable.yaml > bundle.svg` and open the SVG in the browser. Check the service icons are still circles and that there is no circle and clippath inside each service group.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/jujusvg/64/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/jujusvg/67/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/jujusvg/68/
Test PASSed.

File: canvas.go [hunk 1]
```diff
@@ -248,6 +237,20 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
 	}
 }
 
+func (c *Canvas) iconClipPath(canvas *svg.SVG) {
+	canvas.Circle(
+		serviceBlockSize/2-iconSize/2+5, // for these two, add an offset to help
+		serviceBlockSize/2-iconSize/2+7, // hide the embossed border.
+		serviceBlockSize/4,
+		`id="service-icon-mask" fill="none"`)
+	canvas.ClipPath(`id="clip-mask"`)
```
Comment (makyo) [inline canvas.go:246]: `defer canvas.ClipEnd()` is more in the Go style

File: canvas.go [hunk 2]
```diff
@@ -248,6 +237,20 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
...
+	canvas.Use(
+		0,
+		0,
+		`#service-icon-mask`)
+	canvas.ClipEnd()
```
Comment (makyo) [inline canvas.go]: If you defer, you can remove this

File: canvas.go [hunk 3]
```diff
@@ -248,6 +237,20 @@ func (c *Canvas) servicesGroup(canvas *svg.SVG) {
 	}
 }
 
+func (c *Canvas) iconClipPath(canvas *svg.SVG) {
```
Comment (makyo) [inline canvas.go:240]: This method should have its own tests to ensure that we know the ramifications of changing serviceBlockSize or iconSize

**Comment by makyo** [comment]
:+1: with a small change and a test, QA okay!

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/jujusvg/69/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/jujusvg/70/
Test PASSed.

**Comment by anthonydillon** [comment]
Thanks :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/jujusvg-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/jujusvg-merge/23

**Comment by anthonydillon** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/jujusvg-merge

**Pull request merged**